FIX: update documentation to use cleaner furo theme and add markdown support via myst

lwasser opened the pull request:
closes #276 
This pr does a bunch of cleanup things

* It adds support for myst which allows us to include markdown in our docs for easier contributing! 
* It cleans up the landing page of the docs to make it easier for users to find content
* It embeds the CONTRIBUTING.md file into the documentation for easier discovery
* It just cleans up extra stuff from the conf.py file that we don't need! 
* I finessed the documentation styles so its just a bit cleaner as well :) 

In case you don't want to build locally i've added some screenshots of what the new theme and landing page look like! 

# Documentation landing page
(I might make the grid cards grey or add more of a drop shadow at some point) 

<img width="831" alt="Screen Shot 2022-11-26 at 11 37 22 AM" src="https://user-images.githubusercontent.com/7649194/204104262-91be80ac-7148-413f-8f87-6c7eda1e325c.png">

## Landing page lower down 
<img width="828" alt="Screen Shot 2022-11-26 at 11 37 29 AM" src="https://user-images.githubusercontent.com/7649194/204104259-1b0be103-b7cc-45c0-8643-1ff8409ee809.png">

## New sidebar navigation 

so nav is not split between top and side which i think is confusing .
<img width="269" alt="Screen Shot 2022-11-26 at 11 37 11 AM" src="https://user-images.githubusercontent.com/7649194/204104264-372b9caf-1e45-4671-94c9-0b82f6348210.png">

lwasser commented on docs/conf.py:
```diff
@@ -88,55 +79,22 @@
+# Theme and css
+html_theme = "furo"
+html_static_path = ["_static"]
+html_css_files = ["stravalib.css"]
```
This just adds the custom css to clean up the look of our docs!

jsamoocha replied:
Ok, I think the .css sets a fixed font color that creates the issue in dark mode. See my inline comment.

lwasser replied:
@jsamoocha yes great catch. i hadn't tested on dark mode. I just pushed a fix that removes all color settings from the CSS. i keep forgetting about dark mode options in these themes. 

<img width="1350" alt="Screen Shot 2022-11-29 at 8 24 04 AM" src="https://user-images.githubusercontent.com/7649194/204570143-bfdb5b42-48b2-4ecb-bc03-586fcb93e1ac.png">

jsamoocha replied:
It was pure coincidence that I found that dark mode issue - just decided to try `make -C docs serve` after sunset 😁

lwasser commented on docs/contributing/how-to-contribute.md:
````diff
@@ -0,0 +1,3 @@
+
+
+```{include} ../../CONTRIBUTING.md
````
this is the embed to contributing.md is in the documentation too!

lwasser commented on docs/index.md:
```diff
@@ -0,0 +1,121 @@
+# Welcome to the Stravalib Documentation!
+
+::::{grid} 2
```
this makes the landing page prettier / easier for a user to navigate

lwasser replied:
<img width="831" alt="Screen Shot 2022-11-26 at 11 37 22 AM" src="https://user-images.githubusercontent.com/7649194/204104406-481e7363-5860-4c38-b5ed-45e31b1ae5bd.png">

lwasser commented on docs/index.md:
````diff
@@ -0,0 +1,121 @@
+*  and more intelligent error handling.
+
+
+```{toctree}
````
This is the hidden navigation that the user sees on all pages on the left hand navigation bar

<img width="269" alt="Screen Shot 2022-11-26 at 11 37 11 AM" src="https://user-images.githubusercontent.com/7649194/204104389-da41274b-ba36-401d-8a48-5305baf44c85.png">

lwasser commented:
The last PR will add a development guide. i see docs are failing so let me look into that now.

lwasser commented:
ok all passing now and I squashed down commits! ready for review!

jsamoocha commented on docs/_static/stravalib.css:
```diff
@@ -0,0 +1,48 @@
+}
+
+body p {
+    color: #222;
```
Maybe we should not hard-code colors here? These seem to lead to readability issues in dark mode. Maybe it's better (and easier to maintain) to let the theme decide about colors and/or font sizes to prevent any potential accessibility issues?

jsamoocha reviewed (commented):
Thanks for this contribution. The documentation starts to look good! Being able to use markdown is a great improvement.

There is an accessibility issue with the custom .css, please see my other comments.

There is also a problem rendering the changelog:

<img width="804" alt="Notification_Center" src="https://user-images.githubusercontent.com/1788027/204373553-6a12b7d4-856c-48a6-9af7-fe57d39f9481.png">

I'm also missing the latest entries to the changelog. Can it be that it's not correctly included/embedded?

lwasser commented on changelog.md:
```diff
@@ -90,42 +90,35 @@
 
 ## 0.6.3
------
 * Fixed update_activity to include description (@hozn, #91)
```
removed rst that i must have missed in that other big pr

jsamoocha commented on docs/contributing/documentation-changelog.md:
````diff
@@ -0,0 +1,3 @@
+
+
+```{include} ../../changelog.md
````
I pulled your recent changes locally, but the changelog seems to have disappeared completely. Am I missing something?

<img width="291" alt="Stravalib_Docs" src="https://user-images.githubusercontent.com/1788027/204824650-f4e81f3e-6e6c-4a0e-ba68-ad8971363fce.png">

lwasser replied:
oh gosh. Let me look again locally. I thought i had fixed that. 

There is a "feature" / "bug" where if there are two h1 tags on a page, it won't render in the TOC.  BUT there is no header in that page at all purposefully to address that bug. Let me see if it needs a different approach. thank you again for catching all of these issues. goodness.

lwasser commented on docs/index.md:
```diff
@@ -0,0 +1,121 @@
+:maxdepth: 2
+
+How to Contribute <contributing/how-to-contribute>
+Change Log <changelog>
```
Oh i see the issue @jsamoocha i bet my local build needed a clean and update. that link needs to go to:

contributing/documentation-changelog

This is a leah-error for sure.

lwasser replied:
```suggestion
Change Log <contributing/documentation-changelog>
```

lwasser commented on docs/index.md:
```diff
@@ -109,7 +109,7 @@ Athletes <get-started/athletes>
 
 How to Contribute <contributing/how-to-contribute>
-Change Log <changelog>
+Change Log <contributing/documentation-changelog>
```
I will test this locally @jsamoocha in hopes of saving you time :) my goodness

lwasser commented:
ok i built this locally. funny enough what happened was i fixed the index file and the docs were building fine locally. but i never pushed or committed the change to GitHub. it should be clean now @jsamoocha i hope!!  ive looked at it in dark and regular mode too. i see the changelog in the side bar. hopefully it works for you as well.

lwasser commented:
there is one quirk in the PR that i'm seeing that should get fixed with the other open  draft PR https://github.com/stravalib/stravalib/pull/278 it's failing because it needs this PR with the requirements update. the full side bar should work and be cleaner with that PR in place so once we merge this i'll work on that one! (assuming the changelog works now ) and styles are readable for you!

jsamoocha commented:
Ok, almost there. I noticed a broken link (to the newly added code of conduct). It then occurred to me that the linkcheck target is not part of the doc build anymore and therefore didn't catch this. When adding and building with linkcheck locally, there are 2 more broken links in the docs.

Maybe we should make sure that all doc build targets depend on linkcheck?

lwasser commented:
@jsamoocha i'm happy to add that and am working on it locally before i start my day job :)  is see those errors now. i'll add this either at lunch time or this evening as i want to rethink the makefile just a little bit with this change. I think there will be one broken link on the contributing file that is to a file in https://github.com/stravalib/stravalib/pull/278 that PR was built upon this pr and depends on this pr. I could simply remove the link and will just do that in this pr and re-add it in the draft pr. I just didn't want to have a GINORMOUS pr for you to review it's so much harder. here we've been able to go back and forth on a lot of things that i think were easier to catch with a smaller pr. but i see there are some other broken links too so i'll work on those now locally and add link check back to the build! great catch!

lwasser commented:
this is NOT READY to review - will fix links later today. but hopefully the CI runs as we want it to.

lwasser commented on .github/workflows/build-docs.yml:
```diff
@@ -36,4 +36,7 @@ jobs:
           python3 -m pip install -r ./requirements.txt
           python3 -m pip install .
-      - run: make -C docs html
+      - name: Build docs & linkcheck
```
i added linkcheck to the vanilla make docs call here.

lwasser commented on CONTRIBUTING.md:
```diff
@@ -22,12 +22,12 @@ value contributions but also value respectful interactions with stravalib users.
 **Please be considerate and respectful of others** in all of your communications
 in this repository.
-Everyone must abide by our [Code of Conduct](https://www.github.com/stravalib/stravalib/CODE_OF_CONDUCT.md). Please
+Everyone must abide by our [Code of Conduct](https://github.com/stravalib/stravalib/blob/master/CODE_OF_CONDUCT.md). Please
```
i may embed the COC  into the docs for cleaner reading in another PR but not here.

lwasser commented on docs/Makefile:
```diff
@@ -18,26 +18,34 @@ ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees  $(SPHINXOPTS) .
 	@echo "Build finished. The HTML pages are in $(BUILDDIR)/html."
 
+linkcheck: html
+	$(SPHINXBUILD) -b linkcheck $(ALLSPHINXOPTS) $(BUILDDIR)/linkcheck
```
link check now will produce an output file output.txt with all broken things ... much easier to read as well vs looking at the CLI output of sphinx/linkcheck

lwasser commented on docs/Makefile:
```diff
@@ -18,26 +18,34 @@ ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees  $(SPHINXOPTS) .
 clean:
 	rm -rf $(BUILDDIR)/html/*
 	rm -rf $(BUILDDIR)/doctrees
+	rm -rf $(BUILDDIR)/linkcheck
```
so remove that dir in case you wish to do a full clean of docs

lwasser commented:
PHEW! ok this is all ✅ so we're good for another review. it looks ok locally. i've fixed a bunch of broken and redirected links. And updated the makefile accordingly. also found a changelog item that i must have added that hasn't happened yet :) removed that even tho the psychic in me knows it will happen in #278 once it's merged :) and i fixed a toctree bug that was making a doc page render in the side bar weirdly too!

jsamoocha reviewed (approved):
LGTM, great work!

lwasser commented:
sweet merging. thanks for the careful reviews @jsamoocha !!
